src/StructuredLogger/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -250,10 +290,17 @@ private void Write(BuildStartedEventArgs e)
             }
             else
             {
-                Write(0);
+                Write(e.BuildEnvironment?.Where(kvp => IsWellKnownEnvironmentDerivedProperty(kvp.Key)));
             }
         }
 
+        private static bool IsWellKnownEnvironmentDerivedProperty(string propertyName)
+        {
+            return propertyName.StartsWith("MSBUILD") ||

```

Comment:
Should these use StringComparison.OrdinalIgnoreCase? Or nah?

Reply:
Nah :-)
It's same in MSBuild

https://github.com/dotnet/msbuild/blob/5d1509792899a9ac25f59b8b0034dd98989a30b1/src/Shared/EnvironmentUtilities.cs#L16-L21

```csharp
        public static bool IsWellKnownEnvironmentDerivedProperty(string propertyName)
        {
            return propertyName.StartsWith("MSBUILD") ||
                propertyName.StartsWith("COMPLUS_") ||
                propertyName.StartsWith("DOTNET_");
        }
```